src/wp-includes/media.php

```diff
@@ -5640,6 +5670,20 @@ function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 		return $loading_attrs;
 	}
 
+	/*
+	 * Add `decoding` with a value of "async" for every image unless it has a
+	 * conflicting `decoding` attribute already present.
+	 */
+	if ( 'img' === $tag_name ) {
+		if ( isset( $attr['decoding'] ) ) {
```

Comment:
@felixarntz @mukeshpanchal27 

I believe there are some fundamental differences between `isset()` and `array_key_exists()` that are key here. As it stands, you'll see that `test_wp_get_attachment_image_decoding_attr()` will fail when testing `null`. The `isset` will fail this check, and the function will return the default `async`. So whenever we call `wp_get_loading_optimization_attributes` passing `null` for `decoding`, we'll end up with the `decoding="async"`.

Reply:
Thank you @pereirinha, that is a great catch.

After further review, I think it is okay to ignore that failure and remove the `null` case specifically from the `data_wp_get_attachment_image_decoding_attr()` data provider. While that is technically a breaking change, I think it's okay because:
* The parameter is not documented to support `null`.
* It is unlikely somebody is actually passing that value, see e.g. https://wpdirectory.net/search/01HAQF6ZXQMZEJY4EBEDK3B4GM.
* Instead of `null`, `false` can be used to achieve the same behavior, and that's also the documented value.
* `isset()` is faster than `array_key_exists()`, so there's a good reason to not support `null` and interpret as "not existing" instead.